// @clerk/clerk-expo/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var src_exports = {};
__export(src_exports, {
  Clerk: () => import_singleton.clerk,
  ClerkLoaded: () => import_clerk_react.ClerkLoaded,
  ClerkLoading: () => import_clerk_react.ClerkLoading,
  EmailLinkErrorCode: () => import_clerk_react.EmailLinkErrorCode,
  MagicLinkErrorCode: () => import_clerk_react.MagicLinkErrorCode,
  MultisessionAppSupport: () => import_clerk_react.MultisessionAppSupport,
  SignedIn: () => import_clerk_react.SignedIn,
  SignedOut: () => import_clerk_react.SignedOut,
  WithClerk: () => import_clerk_react.WithClerk,
  WithSession: () => import_clerk_react.WithSession,
  WithUser: () => import_clerk_react.WithUser,
  isClerkAPIResponseError: () => import_clerk_react.isClerkAPIResponseError,
  isEmailLinkError: () => import_clerk_react.isEmailLinkError,
  isKnownError: () => import_clerk_react.isKnownError,
  isMagicLinkError: () => import_clerk_react.isMagicLinkError,
  isMetamaskError: () => import_clerk_react.isMetamaskError,
  useAuth: () => import_clerk_react.useAuth,
  useClerk: () => import_clerk_react.useClerk,
  useEmailLink: () => import_clerk_react.useEmailLink,
  useMagicLink: () => import_clerk_react.useMagicLink,
  useOrganization: () => import_clerk_react.useOrganization,
  useOrganizationList: () => import_clerk_react.useOrganizationList,
  useOrganizations: () => import_clerk_react.useOrganizations,
  useSession: () => import_clerk_react.useSession,
  useSessionList: () => import_clerk_react.useSessionList,
  useSignIn: () => import_clerk_react.useSignIn,
  useSignUp: () => import_clerk_react.useSignUp,
  useUser: () => import_clerk_react.useUser,
  withClerk: () => import_clerk_react.withClerk,
  withSession: () => import_clerk_react.withSession,
  withUser: () => import_clerk_react.withUser
});
module.exports = __toCommonJS(src_exports);
var import_clerk_react = require("@clerk/clerk-react");
var import_singleton = require("./singleton");
__reExport(src_exports, require("./ClerkProvider"), module.exports);
__reExport(src_exports, require("./useOAuth"), module.exports);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Clerk,
  ClerkLoaded,
  ClerkLoading,
  EmailLinkErrorCode,
  MagicLinkErrorCode,
  MultisessionAppSupport,
  SignedIn,
  SignedOut,
  WithClerk,
  WithSession,
  WithUser,
  isClerkAPIResponseError,
  isEmailLinkError,
  isKnownError,
  isMagicLinkError,
  isMetamaskError,
  useAuth,
  useClerk,
  useEmailLink,
  useMagicLink,
  useOrganization,
  useOrganizationList,
  useOrganizations,
  useSession,
  useSessionList,
  useSignIn,
  useSignUp,
  useUser,
  withClerk,
  withSession,
  withUser,
  ...require("./ClerkProvider"),
  ...require("./useOAuth")
});
